### PR TITLE
Aitboudad ssr lazyload

### PR DESCRIPTION
**Pull request by christopherthielen**
Based on https://github.com/ui-router/angular/pull/127

@aitboudad Please take a look.

### CODE REVIEW

File: src/uiRouterNgModule.ts [hunk 1]
```diff
@@ -172,6 +187,15 @@ export interface RootModule extends StatesModule {
...
   deferIntercept?: boolean;
+
+  /**
+   * Tells Angular to defer first render until after the initial transition is complete.
+   *
+   * When `true`, adds an `APP_INITIALIZER` which resolves on any `onSuccess` or `onError`.
+   *
+   * Defaults to `false`
+   */
+  deferInitialRender?: boolean;
```
Comment (aitboudad) [inline src/uiRouterNgModule.ts]: FYI angular router use [initialNavigation](
https://github.com/angular/angular/blob/4.4.x/packages/router/src/router_module.ts#L259) option

**Comment by aitboudad** [comment]
Test on my side and it's work as expected but some changes on [baseLocationService](https://github.com/ui-router/core/blob/master/src/vanilla/baseLocationService.ts) using typeof self|Event !== 'undefined':
- [self](https://github.com/ui-router/core/blob/master/src/vanilla/baseLocationService.ts#L19) is not defined
- [Event](https://github.com/ui-router/core/blob/master/src/vanilla/utils.ts#L74) is not defined

**Comment by christopherthielen** [comment]
@aitboudad in what environment are`self` and `Event` not defined?

**Comment by aitboudad** [comment]
@christopherthielen in nodejs env

**Comment by aitboudad** [comment]
also https://github.com/ui-router/core/blob/master/src/vanilla/browserLocationConfig.ts may need some little changes for location, I used https://github.com/ui-router/angular/issues/134#issuecomment-310678951 as a workaround ;)

**Comment by christopherthielen** [comment]
ah.  I didn't know that.

I've made some (unreleased) changes that I think will address this.  Would you mind building `core` and testing against browser and server-side?

https://github.com/ui-router/core/commit/2d206ba3dac0398ac2692d91b49b9628f09f4104

**Comment by christopherthielen** [comment]
@aitboudad I released the core changes as 5.0.10.  I think this is ready to go, but would like you to confirm.  If I don't hear from you by tomorrow, I'm just going to merge and hope for the best :) . I'd also like to release a new version of `uirouter/angular`.

**Comment by aitboudad** [comment]
I can confirm now.
#134 is fixed too, thanks!

**Pull request merged**